addons/account_edi/models/account_move.py

```diff
@@ -1,25 +1,36 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, fields
+from odoo import api, fields, models
 
 
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    edi_document_ids = fields.One2many('ir.attachment', 'res_id', domain=[('res_model', '=', 'account.move'), ('edi_format_id', '!=', False)])
+    edi_document_ids = fields.One2many(comodel_name='account.edi.document', inverse_name='move_id')
+    edi_error = fields.Text(compute='_compute_edi_error')
+
+    @api.depends('edi_document_ids', 'edi_document_ids.error')
+    def _compute_edi_error(self):
+        for move in self:
+            move.edi_error = '\n'.join(["%s: %s" % (d.edi_format_id.name, d.error) for d in move.edi_document_ids.filtered(lambda d: d.error)[:3]])
+            print("df", len(move.edi_document_ids.filtered(lambda d: d.error)), move.edi_error)
 
     def post(self):
         # OVERRIDE
         # Generate the electronic documents for the move.
-        existing_attachments = self.env['ir.attachment'].search([
-            ('res_model', '=', 'account.move'),
-            ('res_id', 'in', self.ids),
-            ('edi_format_id', 'in', self.journal_id.edi_format_ids.ids)])
-        existing_attachments.unlink()
         res = super(AccountMove, self).post()
+
         for move in self:
-            move.journal_id.edi_format_ids._create_ir_attachments(move)
+            documents = move.journal_id.edi_format_ids._create_edi_document(move)
+            documents.write({'state': 'to_send'})
+
+        return res
```

Comment:
can we make `_create_edi_document` is only creating the edi.document and have a method like `documents._proceed()` that will contains all the edi logic like:
```
def _proceed_synchronously(self):
  if self.edi_format_id._needs_web_services():
    return

  if self.state == 'to_send':
    ...
    if success:
     self.state = False
  elif self.state == 'to_cancel':
    ...

def _proceed_asynchronously(self):
  if not self.edi_format_id._needs_web_services():
    return

  ...
    
```